pkg/metrics/cluster/config_watcher.go

```diff
@@ -131,6 +136,11 @@ func (w *configWatcher) Refresh(ctx context.Context) (err error) {
 		reshardDuration.WithLabelValues(success).Observe(time.Since(start).Seconds())
 	}()
 
+	// This is used to determine if the context was already exceeded before calling the kv provider
+	if err = ctx.Err(); err != nil {
+		return fmt.Errorf("context deadline exceeded before calling store.all")
+	}
```

Comment:
There's a chance that the context will cancel right after this check but still before `w.store.All` gets invoked. Can you supplant this with a debug-level log that happens after w.store.All?